### PR TITLE
Feat/array-loop

### PR DESCRIPTION
**Pull request by gijsvanwichen**

### CODE REVIEW

File: __tests__/arrayLoop/1.0/function.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      "meta": {
+        "type": "Output",
+        "output": {
+          "type": "Number",
+          "scoped": true
+        }
+      },
+      "name": "index",
+      "label": "Index",
+      "info": "The name for the variable under which the current iteration will be available. Starts at 0. For example, in the 1st iteration this variable holds the value 0 and in 10th iteration, this variable holds the value 9."
```
Comment (jwielinga) [inline __tests__/arrayLoop/1.0/function.json:48]: +1 for using indexing at 0

File: functions/arrayLoop/1.0/index.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/* eslint-disable no-await-in-loop */
+const loop = async ({ array }, steps) => {
```
Comment (jwielinga) [inline functions/arrayLoop/1.0/index.js:2]: What if the value is null? You can handle this via: `({ array = [] }, steps) => {`

File: __tests__/arrayLoop/1.0/function.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
+{
```
Comment (jwielinga) [inline __tests__/arrayLoop/1.0/function.json:1]: This should not be in tests right?

**Review by jwielinga — CHANGES_REQUESTED** [review]

**Review by jwielinga — DISMISSED** [review]
Not to test about the steps I guess :)

File: functions/arrayLoop/1.0/index.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+/* eslint-disable no-await-in-loop */
+const loop = async ({ array = [], outputType }, steps) => {
+  for (let index = 0; index < array.length; index += 1) {
```
Comment (jwielinga) [inline functions/arrayLoop/1.0/index.js:3]: Change to index++

**Review by jwielinga — CHANGES_REQUESTED** [review]
See comments

File: functions/arrayLoop/1.0/index.js (hunk 4 above)
Comment (gijsvanwichen) [reply functions/arrayLoop/1.0/index.js:3]: I use the same method as in the native loop function: https://github.com/bettyblocks/native-functions/blob/master/functions/loop/1.0/index.js @jwielinga

**Pull request closed without merging**